### PR TITLE
improve laggy subset finalization

### LINKED ISSUE #3164: Subsets taking a long time to finalize in Cubeviz on main
See the video below - in Cubeviz (I confirmed that Imviz is behaving normally), subsets are taking up to 5-6 seconds to finalize when drawn, which is rather annoying. I haven't had a chance to track down why this might be happening.

https://github.com/user-attachments/assets/0b44ff60-3079-4dcd-bcee-840dee406e2e

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request limits the updates to `subset_selected_changed_callback` (originally introduced in #2939) to only trigger for subset update messages corresponding to the currently selected subset (and only when the subset state has been changed, not color, etc) and also minimizes spectral extraction calls when `subset_select.items` are changed to only affect the selected entry.

~Fixes #3164~

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/app.py [hunk 1]
```diff
@@ -614,12 +614,14 @@ def _on_layers_changed(self, msg):
...
 
-        self.state.layer_icons = {
-            **self.state.layer_icons,
-            **child_layer_icons
-        }
+        if child_layer_icons:
+            self.state.layer_icons = {
+                **self.state.layer_icons,
+                **child_layer_icons
+            }
```
Comment (kecnry) [inline jdaviz/app.py:624]: FYI: @bmorris3 - does this seem reasonable?  I'm trying to avoid an update to the state again if no new child layers are added.

**Review by javerbukh — APPROVED** [review]
I think I notice a marginal improvement on my windows PC and the code looks reasonable so I will approve. Looking forward to more optimization in this direction!

**Comment by rosteen** [comment]
I found this to be a very minor improvement, but this apparently isn't the source of most of the (millions of) calls to `startswith` that I was seeing. I think we can get it in, but let's not close the issue that started it (I'll have to remake that issue, it got deleted).

**Review by rosteen — APPROVED** [review]
Approved, but please don't close until @pllim has remade the deleted corresponding issue.

**Comment by kecnry** [comment]
I won't have this close #3181 since it doesn't fully address the lag (although it _should_ address the unnecessary calls to spectral extraction during the subset creation process).

**Pull request merged**